File: 0368-largest-divisible-subset/0368-largest-divisible-subset.cpp
class Solution {
public:
    vector<int> largestDivisibleSubset(vector<int>& nums) {
        int n = nums.size();
        std::sort(nums.begin(), nums.end());
        vector<vector<int>> dp;
        vector<int> ans;
        for(int i=0;i<n;i++){
            dp.push_back({nums[i]}); 
            for(int j=0;j<i;j++){
                if(nums[i]%nums[j]==0 && dp[j].size()>=dp[i].size()){
                    dp[i]=dp[j];
                    dp[i].push_back(nums[i]);
                }
            }
            if(dp[i].size()>ans.size()){
                ans=dp[i];
            }
        }
        
        return ans;
    }
};

File: 0368-largest-divisible-subset/0368-largest-divisible-subset.py
class Solution:
    def largestDivisibleSubset(self, nums: List[int]) -> List[int]:
        nums.sort()
        dp = [[nums[i]] for i in range(len(nums))]
        for i in range(len(nums)):
            for j in range(i):
                if nums[i]%nums[j]==0 and len(dp[j])>=len(dp[i]):
                    dp[i] = dp[j][:]
                    dp[i].append(nums[i])
        ans = []
        for i in range(len(dp)):
            if len(dp[i])>len(ans):
                ans = dp[i]
        return ans